lib/sorcery/controller/submodules/jwt_auth.rb

```diff
@@ -0,0 +1,75 @@
+module Sorcery
+  module Controller
+    module Submodules
+      module JwtAuth
+        def self.included(base)
+          base.send(:include, InstanceMethods)
+        end
+
+        module InstanceMethods
+          # This method return generated token if user can be authenticated
+          def jwt_auth(*credentials)
+            user = user_class.authenticate(*credentials)
+            if user
+              user_params = Config.jwt_user_params.each_with_object({}) do |val, acc|
+                acc[val] = user.public_send(val)
+              end
+              { Config.jwt_user_data_key => user_params,
+                Config.jwt_auth_token_key => jwt_encode(user_params) }
+            end
+          end
+
+          # To be used as a before_action.
+          def jwt_require_auth
+            jwt_not_authenticated && return unless jwt_user_id
+
+            @current_user = Config.jwt_set_user ? User.find(jwt_user_id) : jwt_user_data
+          rescue JWT::VerificationError, JWT::DecodeError
+            jwt_not_authenticated && return
+          end
+
+          # This method creating JWT token by payload
+          def jwt_encode(payload)
+            JWT.encode(payload, Config.jwt_secret_key)
```

Comment:
You should specify an algorithm to use, else the created JWT is unsigned!

See here: https://github.com/jwt/ruby-jwt#algorithms-and-usage

I think it should be a config option, with `"HS256"` as its default value.

The same algorithm parameter has to be passed to `JWT.decode` as well.